docs/versioning-policy.md

```diff
@@ -0,0 +1,70 @@
+# Upjet provider versioning and upgrade policy
+
+This policy recommends how Upjet providers should implement versioning numbers and handle upgrades to new versions.
+
+## Versioning 
+
+Upjet-based providers **MUST** follow the [semantic versioning](https://semver.org/) numbering scheme. 
+
+Specifically, a version contains 3 digits known as `major`, `minor` and `patch` numbers in the format `major.minor.patch` (e.g., `v1.2.3`).
+
+### Versioning specification
+
+1. The `major` version number **MUST** be incremented if the `major` version number of the Terraform provider it is generated from is incremented.
+2. The `minor` version number **MUST** be incremented when new functionality is released or when [unavoidable breaking changes](#unavoidable-breaking-changes) are introduced. 
+3. The `patch` version number **SHOULD ONLY** be incremented for a release containing **ONLY** bug fixes and no new features.
+4. A release that increments the `minor` or `patch` version **ONLY**, **MUST** be backward compatible.
```

Comment:
This is in conflict with the unavoidable breaking changes caveat from point number 2.